corehq/apps/app_manager/suite_xml/sections/endpoints.py

```diff
@@ -0,0 +1,72 @@
+from typing import List
+
+from corehq.apps.app_manager import id_strings
+from corehq.apps.app_manager.suite_xml.contributors import (
+    SuiteContributorByModule,
+)
+from corehq.apps.app_manager.suite_xml.xml_models import (
+    Argument,
+    PushFrame,
+    SessionEndpoint,
+    Stack, StackDatum,
+)
+from corehq.apps.app_manager.xpath import XPath, session_var
+
+
+class SessionEndpointContributor(SuiteContributorByModule):
+
+    def get_module_contributions(self, module) -> List[SessionEndpoint]:
+        endpoints = []
+        if module.session_endpoint_ids:
+            endpoints.append(_get_module_endpoint(module))
+        for form in module.get_suite_forms():
+            if form.session_endpoint_ids:
+                endpoints.append(_get_form_endpoint(form))
+        return endpoints
+
+
+def _get_module_endpoint(module) -> SessionEndpoint:
+    id_string = id_strings.case_list_command(module)
+    return SessionEndpoint(
+        arguments=[Argument(id=id_) for id_ in module.session_endpoint_ids],
+        stack=_build_stack(id_string, module.session_endpoint_ids),
+    )
+
+
+def _get_form_endpoint(form) -> SessionEndpoint:
+    id_string = id_strings.form_command(form)
+    return SessionEndpoint(
+        arguments=[Argument(id=id_) for id_ in form.session_endpoint_ids],
+        stack=_build_stack(id_string, form.session_endpoint_ids),
+    )
+
+
+def _build_stack(id_string: str, endpoint_ids: List[str]) -> Stack:
+    """
+    Returns a stack with the command for a form or case list, and a list
+    of datums whose ID and session variable name are both the element ID.
+
+    .. note::
+       This assumes that the session variable name and the datum ID
+       should be the same.
+
+    e.g. ::
+
+        <stack>
+          <push>
+            <command value="'m1-f0'"/>
+            <datum id="case_id" value="instance('commcaresession')/session/data/case_id"/>
+          </push>
+        </stack>
+
+    """
+    stack = Stack()
+    frame = PushFrame()
+    frame.add_command(XPath.string(id_string))
+    for endpoint_id in endpoint_ids:
+        frame.add_datum(StackDatum(
+            id=endpoint_id,
+            value=session_var(endpoint_id))
```

Comment:
I'm maybe not completely understanding how this works, but how is this session variable populated in the reminders app workflow? Doesn't that mean that a user will have had to already navigate to the case they want in CommCare before navigating the reminders app? Or is there some other way this is populated? 

The spec seems to suggest using an `<argument>` whose name is the value here, which I assume is something that the reminders app would pass in when this callout is triggered. 
```
<argument id=”person_case_id”/>
...
<datum id=”case_id” value=”$person_case_id”>
````

Reply:
that sounds in line with Shubham's comment [here](https://dimagi-dev.atlassian.net/browse/SC-953?focusedCommentId=115731)

Reply:
Yeah. This code is wrong. Shubham's comment is spot on, and clarifies the problem, and how to fix it. We need to have an `<argument>` for each case the form needs.